### DISCUSSION TITLE
Serve static files dynamically in development

### DISCUSSION THREAD

**Discussion by alberteddu** [vercel/next.js#16570] [Help]
Hello,

I imagine my use case is a bit strange but I would really like to be able to do this with Next.js as I really like it.
I would like to serve some dynamic files by adding a get route (maybe to the webpack dev server?) in development. Because of my usecase, these files cannot be in the public directory at this point. 

When deploying to production, I will manually copy the files that I need to public before building, so the handler is not needed anymore.

Is there a simple way to achieve this? It would be OK to use a custom server but in that case I would like to avoid losing any of the features I get with the default one. Also, in production the handler would not really be needed, so I am unsure what to do.

I would be grateful for any tip!

Thanks,
Alberto

**Comment by alberteddu** [comment]
What I did was create a custom server that does the dynamic serving of the static files. `yarn dev` will use this server.
I slightly altered `yarn build` so that it will copy some files to the static folder before building. And I left `yarn start` untouched which should make it behave as usual when deploying to Vercel.

So if I'm not mistaken, this should achieve exactly what I'm looking for. 

Still, I will welcome any feedback about this approach.

Best,
Alberto

**Reply by marcin-codelabs** [reply]
Can you show me your solution attaching files (with a custom server that does the dynamic serving of the static files)?

**Reply by alberteddu** [reply]
Hey, maybe this can help? https://gist.github.com/alberteddu/a6222fe76666c9d42d3391d9255ffe64
You're probably interested in the `server.js` file.
